graphql-compiler-plugin/src/main/java/io/ballerina/stdlib/graphql/compiler/GraphqlFunctionValidator.java

```diff
@@ -0,0 +1,215 @@
+            FunctionDefinitionNode functionDefinitionNode = (FunctionDefinitionNode) node;
+            if (functionDefinitionNode.kind() == SyntaxKind.RESOURCE_ACCESSOR_DEFINITION) {
+                // resource functions are valid - validate function signature
+                validateResourceFunction(functionDefinitionNode, context);
+            } else if (functionDefinitionNode.kind() == SyntaxKind.OBJECT_METHOD_DEFINITION) {
+                // object methods are valid, object methods that are remote functions are invalid
+                if (PluginUtils.isRemoteFunction(context, functionDefinitionNode)) {
+                    context.reportDiagnostic(PluginUtils.getDiagnostic(CompilationErrors.INVALID_FUNCTION,
+                            DiagnosticSeverity.ERROR, functionDefinitionNode.location()));
+                }
+            }
+        }
+    }
+
+    private void validateResourceFunction(FunctionDefinitionNode functionDefinitionNode,
+                                          SyntaxNodeAnalysisContext context) {
+        validateResourceAccessorName(functionDefinitionNode, context);
+        validateReturnType(functionDefinitionNode, context);
+        validateInputParamType(functionDefinitionNode, context);
+    }
+
+    private void validateResourceAccessorName(FunctionDefinitionNode functionDefinitionNode,
+                                              SyntaxNodeAnalysisContext context) {
+        SemanticModel semanticModel = context.semanticModel();
+        Optional<Symbol> symbolOptional = semanticModel.symbol(functionDefinitionNode);
+        if (symbolOptional.isPresent()) {
+            ResourceMethodSymbol resourceMethodSymbol = (ResourceMethodSymbol) symbolOptional.get();
+            if (resourceMethodSymbol.getName().isPresent()) {
+                if (!resourceMethodSymbol.getName().get().equals(PluginConstants.RESOURCE_FUNCTION_GET)) {
+                    context.reportDiagnostic(PluginUtils.getDiagnostic(
+                            CompilationErrors.INVALID_RESOURCE_FUNCTION_ACCESSOR,
+                            DiagnosticSeverity.ERROR, functionDefinitionNode.location()));
+                }
+            } else {
+                context.reportDiagnostic(PluginUtils.getDiagnostic(
+                        CompilationErrors.INVALID_RESOURCE_FUNCTION_ACCESSOR,
+                        DiagnosticSeverity.ERROR, functionDefinitionNode.location()));
+            }
+        }
+    }
+
+    private void validateReturnType(FunctionDefinitionNode functionDefinitionNode, SyntaxNodeAnalysisContext context) {
+        MethodSymbol methodSymbol = PluginUtils.getMethodSymbol(context, functionDefinitionNode);
+        if (methodSymbol != null) {
+            Optional<TypeSymbol> returnTypeDesc = methodSymbol.typeDescriptor().returnTypeDescriptor();
+            if (returnTypeDesc.isPresent()) {
+                // if return type is a union
+                if (returnTypeDesc.get().typeKind() == TypeDescKind.UNION) {
```

Comment:
How does this work for the union of non-primitive types?

For example, 
```
type Person record {
    string name;
};

type Student record {
    string name;
};

resource function get profile() returns Person|Student {
    //...
}
```

?

(Union of non-primitive types should be allowed, for support union types).

Reply:
What are the types this scenario is invalid for? string, float, int, decimal, boolen?

Reply:
example: 
```
resource function get profile() returns int|string {
    //...
}

```

Reply:
Please review the fix 66389f02031bb2bf90a8e4f66859d226b7318d7e

Reply:
Yes. This should be invalid. 
Also `Person|string` is also invalid. 
`Person|Student` should be valid.